[#187364733] -Ft two factor authentication for sellers

EricNiyo73 opened the pull request:
**What does this PR do?**
This PR introduces the functionality for users, specifically Sellers, to enable two-factor authentication (2FA) for added security when logging into the application.

 **Description of Task to be completed?**
The task involves implementing the logic to allow users to enable 2FA for their accounts. When 2FA is enabled for a user, they should be prompted to provide an additional authentication factor, such as an token and OTP code sent via email, during the login process.

 **How should this be manually tested?**
Attempt to log in to the user account with 2FA enabled.
Verify that the login process prompts for a 2FA token i.e jwt token and OTP code sent via email of a user.
Enter a valid 2FA token and submit the form.

 **Any background context you want to provide?**
Verify that the user is successfully logged in after the 2FA token is validated.
Implementing 2FA adds an extra layer of security to user accounts, especially for users with elevated privileges like Sellers who may have access to sensitive data or functionalities.

 **What are the relevant pivotal tracker/Trello stories?**
[#187364733]
 **Screenshots (if appropriate)**
1) A user try to login with valid credentials, if a userRole is a seller, might get this response.
![image](https://github.com/atlp-rwanda/e-commerce-bitcrafters-bn/assets/106761137/a6fed273-2dbb-4e43-add2-aa00026ad7bd)


2) OTP sent and well received via email.
![image](https://github.com/atlp-rwanda/e-commerce-bitcrafters-bn/assets/106761137/6519e908-4e15-4b50-ad8e-d4df62da9cff)



3) A user if tries to insert OTP  token which is not provided and also that OTP does not corresponded to his/her email 
![image](https://github.com/atlp-rwanda/e-commerce-bitcrafters-bn/assets/106761137/92d3c3fe-ac7d-44a7-aadd-14afd0b742dc)


4) A user finally entered a valid OTP code and authenticated successfully and also token would be generated for login.

![image](https://github.com/atlp-rwanda/e-commerce-bitcrafters-bn/assets/106761137/fc067b50-eccb-4860-b01c-24685e969fc4)

AngeloChristian1 commented on test/jwtConfiguration.test.ts:
```diff
@@ -0,0 +1,85 @@
+/* eslint-disable @typescript-eslint/no-explicit-any */
+/* eslint-disable @typescript-eslint/no-unused-vars */
```
Try to remove these eslint comments

AngeloChristian1 commented on package.json:
```diff
@@ -7,8 +7,8 @@
     "build": "tsc",
-    "test": "nyc mocha --require ts-node/register test/*.ts",
-    "coverage": "NODE_ENV=test nyc --reporter=text --reporter=lcov mocha --require ts-node/register test/*.ts --coverage && nyc report --reporter=text-lcov | coveralls",
+    "test": "nyc mocha --require ts-node/register test/*.ts --timeout 800000 --exit",
```
why did you change the test command?
you can just add the timeout on the individual test you want to time

Furaha-Justine commented on src/assets/emailTemplates/index.ts:
I think you can rename this file to something else for better understanding.

Furaha-Justine commented on src/docs/signupDocs.ts:
Also for better structure of project you can create new file for this doc.

AngeloChristian1 commented on .codeclimate.yml:
can you create a fix PR for this file

AngeloChristian1 commented on src/utils/hashPassword.ts:
there is another similar file in the utils named password.ts, This is not needed

AngeloChristian1 commented on src/utils/hashPassword.ts:
Hey, so there is a similar file in the rebased content named passwords.ts and it is the one we are using

AngeloChristian1 committed:
feature(2FA): Two Factor Authentication for Sellers

feature(2FA): Two Factor Authentication for Sellers

feature(2FA): Two Factor Authentication for Sellers only

feature(2FA): Two Factor Authentication for Sellers only
